riscv/src/compiler.rs

```diff
@@ -1391,6 +1403,24 @@ fn process_instruction(instr: &str, args: &[Argument]) -> Vec<String> {
         // atomic and synchronization
         "fence" | "fence.i" => vec![],
 
+        insn if insn.starts_with("amoadd.w") => {
+            let (rd, rs2, rs1, off) = rrro(args);
```

Comment:
![image](https://github.com/powdr-labs/powdr/assets/504195/0f3c9914-fe8a-4e54-8598-402dc4cd20e1)

Not sure the order is correct here? It would help to keep the register names in the same order as the spec, if I understand the spec correctly, that being `(rs1, rs2, rd, off)`.

Reply:
The order in the binary spec has nothing to do with the order in asm. Got it wrong in another instruction because it seems there is no authoritative documentation, but it was caught by the RISC-V test suite.

This one passes, and the one source I found says it is the way I did, so I am confident:
https://msyksphinz-self.github.io/riscv-isadoc/html/rva.html#amoadd-w

Reply:
ok, makes sense